go-controller/pkg/cni/types.go

```diff
@@ -88,14 +89,21 @@ type PodRequest struct {
 	cancel context.CancelFunc
 }
 
-type cniRequestFunc func(request *PodRequest, podLister corev1listers.PodLister) ([]byte, error)
+type PodCheckMode struct {
```

Comment:
I don't think we need to use a struct type here, we can just pass the bool around